src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -97,7 +110,20 @@ export const Configurator = ({
           >
             <EuiFieldText disabled value={dataset.title} />
           </EuiFormRow>
-          {timeFields && timeFields.length > 0 && (
+          {dataset.type === DEFAULT_DATA.SET_TYPES.INDEX_PATTERN &&
```

Comment:
what if type index? or s3? this will never display correct?

Reply:
Yeah, this should only be displayed for index patterns. Spoke with @ashwin-pc on this one, index patterns are the only type with a time field set, so we should display what it is and not allow any edits.

Reply:
what about cloudwatch and other custom types? i thought we want to filter cloudwatch logs by timestamp

Reply:
This specific label is for index patterns only and shows a disabled `OuiFieldText` that displays the time field associated with the index pattern if there is one. It's only disabled for index patterns because they already have a time field attached upon creation, if specified. If not, there will be an option to select a time field with [this selector](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/5ac93609c7947de5f772b6443e285c9f84de55f8/src/plugins/data/public/ui/dataset_selector/configurator.tsx#L125-L152).

Reply:
The logic here reads wrong. Correct me if my understanding is wrong but shouldnt the logic be like this:

```js
if(languageService.getLanguage(language)?.disableDatePicker) {
  return null
}

if(dataset.type === DEFAULT_DATA.SET_TYPES.INDEX_PATTERN) {
  return DisabledInput
} else {
  return Select
}
```